File: apzkr-pzpi-21-1-volovyi-artem/Task4/apz-pzpi-21-1-volovyi-artem-task4/autocab.client/src/interfaces/trip.ts
import { AddressDto } from "./address";
import { CarInfoDto } from "./car";
import { TripStatus } from "./enums";
import { ServiceInfoDto } from "./service";

export interface CreateTripCommand {
  price: number;
  startAddress?: AddressDto;
  destinationAddress?: AddressDto;
  carId: string;
}

export interface TripInfoDto {
  id: string;
  userId: string;
  tripStatus: TripStatus;
  startDateTime: Date;
  price: number;
  startAddress?: AddressDto;
  destinationAddress?: AddressDto;
  carId: string;
  services?: ServiceInfoDto[];
}

export interface TripFullInfo {
  id: string;
  userId: string;
  tripStatus: TripStatus;
  startDateTime: Date;
  price: number;
  startAddress?: AddressDto;
  destinationAddress?: AddressDto;
  car: CarInfoDto;
  services?: ServiceInfoDto[];
}

export interface CancelOwnTripCommand {
  tripId: string;
}

export interface UpdateTripServicesCommand {
  tripId: string;
  services: string[];
}

File: apzkr-pzpi-21-1-volovyi-artem/Task1-Server/AutoCab.Db/Models/Trip.cs
namespace AutoCab.Db.Models;

public class Trip : Entity
{
    public DateTime StartDateTime { get; set; }
    public DateTime EndDateTime { get; set; }
    public TripStatus Status { get; set; }
    public decimal Price { get; set; }

    public Guid StartAddressId { get; set; }
    public Address? StartAddress { get; set; }

    public Guid DestinationAddressId { get; set; }
    public Address? DestinationAddress { get; set; }

    public Guid CarId { get; set; }
    public Car? Car { get; set; }

    public Guid UserId { get; set; }
    public User? User { get; set; }

    public ICollection<Service>? Services { get; set; }
}

public enum TripStatus
{
    Created,
    InProgress,
    WaitingForPassenger,
    Completed,
    Cancelled
}